src/LoggingExtension.cs

```diff
@@ -50,7 +49,8 @@ public LoggingExtension(ILoggerFactory factory)
 
         protected override void Initialize()
         {
-            Context.Policies.Set(typeof(ILogger),   UnityContainer.All, typeof(ResolveDelegateFactory), (ResolveDelegateFactory)GetResolver);
+            Context.Policies.Set(typeof(ILoggerFactory),   UnityContainer.All, typeof(ResolveDelegateFactory), (ResolveDelegateFactory)GetFactoryResolver);
```

Comment:
Just register instance: `RegisterInstance<ILoggerFactory>(LoggerFactory, InstanceLifetime.Singleton)`

Reply:
I tried that first (`Container.RegisterInstance(LoggerFactory, InstanceLifetime.Singleton)` since that's the way I do it in my own projects and after your comment also `Context.Container.RegisterInstance(LoggerFactory, InstanceLifetime.Singleton)`) but both ways break all the tests. This is the exception message I get:
```
Message: Initialization method Microsoft.Logging.Tests.LoggingFixture.TestSetup threw exception. System.MethodAccessException: Attempt by method 'Unity.Microsoft.Logging.LoggingExtension.Initialize()' to access method 'Unity.Microsoft.Logging.LoggingExtension.get_LoggerFactory()' failed..
```
I got a feeling that I'm missing something obvious...

Reply:
After thinking about it ... your method is correct. LogerFactory could be replaced mid-flight and resolution should reflect it. With instance it is more chalanging